nerfstudio/model_components/renderers.py

```diff
@@ -124,65 +128,68 @@ def get_background_color(
         elif background_color == "last_sample":
             background_color = rgb[..., -1, :]
         elif background_color == "random":
-            background_color = torch.rand_like(rgb[..., -1, :]).to(rgb.device)
+            # background_color = torch.rand_like(rgb[..., -1, :]).to(rgb.device)
+            raise RuntimeError("get_background_color should never be called with background_color='random'")
         elif isinstance(background_color, str) and background_color in colors.COLORS_DICT:
             background_color = colors.COLORS_DICT[background_color].to(rgb.device)
         assert isinstance(background_color, Tensor)
         return background_color
 
-    @classmethod
     def blend_background(
-        cls,
-        rgb: Tensor,
-        opacity: Tensor,
-        background_color: Float[Tensor, "3"],
+        self,
+        image: Tensor,
+        background_color: Optional[Float[Tensor, "3"]] = None,
     ) -> Float[Tensor, "*bs 3"]:
-        """Blends the background color into the image.
+        """Blends the background color into the image if image is RGBA.
+        Otherwise no blending is performed (we assume opacity of 1).
 
         Args:
-            rgb: RGB per pixel.
+            image: RGB/RGBA per pixel.
             opacity: Alpha opacity per pixel.
             background_color: Background color.
 
         Returns:
             Blended RGB.
         """
-        if len(opacity.shape) < len(rgb.shape):
-            opacity = opacity.unsqueeze(-1)
+        if background_color is None:
+            background_color_name = self.background_color
+            if background_color_name == "last_sample" or background_color == "random":
+                background_color_name = "black"
```

Comment:
I find this slightly confusing based on the function name. What about adding a function `blend_background_for_metric_computation`, which does this logic before calling `blend_background`? `blend_background` could then make `background_color` a required argument.

Reply:
I believe the logic here is fine. It is not well defined for random/last_sample anyway. I feel like this is better than adding one more method?

Reply:
sure :)

This is very much personal preference, so I'm happy to leave it like this